docs: Add Mix Install/Docs, Start/Stop ABCI Server guides

ltfschoen opened the pull request.

ltfschoen committed:
docs: Add Mix Installation and Documentation generation guide

ltfschoen committed:
fix: Fixes Issue #4. Update to Ranch 1.4.0 to support GNU Make 4.2.1

Previous version of GNU Make 3.81 supported Ranch 1.3.2, but after upgrading to GNU Make 4.2.1 it did not support Ranch 1.3.2, so it was necessary to upgrade to the latest Ranch 1.4.0

ltfschoen committed:
docs: Update Reamde with Show ABCI Server Module Info, Start/Stop ABCI Server

ltfschoen committed:
docs: Add ABCI-CLI Test to Readme

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,87 @@
+
+* Create a Fake Module named Foo
+  ```
+  iex(1)> defmodule Foo do
````
This module doesn’t implement the `abci_app` behaviour, so it won’t be able to handle ABCI requests.

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,87 @@
+   <<70, ..., 117, ...>>, {:bar, 0}}
+  ```
+
+* Show ABCI Server Information (using `module_info/1` which is the Erlang equivalent of Elixir's `__info__/1`), Start ABCI Server, Stop ABCI Server
````
This is a generic Erlang function. I don’t see much utility in putting information about this in README.

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,87 @@
+  ]
+  ```
+
+  * Run the ABCI Server's `start_listener` and `stop_listener` functions
````
This is already written in `doc/overview.edoc` (though for Erlang).

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,87 @@
+    :ok
+    ```
+
+  * Test the Running ABCI Server (Erlang) in separate Bash Terminal Tab - https://github.com/tendermint/abci#tools
````
This is Tendermint-related documentation, no need to duplicate it here.

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,87 @@
+
+* Documentation Generation. Open Documentation in Web Browser
+  ```bash
+  cd deps/abci_server/ && make docs && open doc/index.html && cd ../../
````
This is already covered in the “Documentation” section.

ltfschoen committed:
fix: Remove proposed change to Ranch 1.4.0 from Makefile

ltfschoen committed:
docs: Add Troubleshooting section with possible Ranch 1.4.0 fix

ltfschoen commented:
Based on feedback from @KrzysiekJ in https://github.com/KrzysiekJ/abci_server/issues/4 I've modified this Pull Request https://github.com/KrzysiekJ/abci_server/pull/3/files and removed the proposed change to the abci_server's Makefile so it still uses Ranch 1.3.2. The Pull Request now only proposes to modify the Readme and includes a Troubleshooting section that may help other users who encounter the IEx error `(Mix) Could not start application ranch: could not find application file: ranch.app` by suggesting that they try adding Ranch 1.4.0 to their Mix dependencies.

KrzysiekJ commented on README.md:
````diff
@@ -2,10 +2,103 @@
+
+* Install Mix Dependencies
+  ```bash
+  mix deps.get
````
This looks like a generic Elixir command, no need to put it here.

KrzysiekJ commented:
Left some comments. Generally I have a feeling that the docs provided are much redundant, not project-specific. If they get fixed (the current example will not work), they may become valuable as a blog post or tutorial or on wiki. For example, adding Erlang packages as dependencies to Elixir projects (and vice versa) is a generic topic and it should be covered perhaps in Elixir documentation.
